src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.xml

```diff
@@ -5,7 +5,20 @@
     xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
-        <service id="annotations.reader" class="Doctrine\Common\Annotations\AnnotationReader" public="false" />
+        <service id="annotations.reader" class="Doctrine\Common\Annotations\AnnotationReader" public="false">
+            <argument>null</argument>
+            <argument type="service">
+                <!-- dummy arg to register class_exists as annotation loader only when required -->
```

Comment:
such dummy arg could cause issue if the library adds an optional arg in the future.
Thus, I think PHP used to do weird things regarding the evaluation of useless constructor arguments.

We should use a configurator instead of using such hack.